src/main/java/com/univocity/parsers/containers/Row.java

```diff
@@ -0,0 +1,156 @@
+package com.univocity.parsers.containers;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Set;
+
+import com.sun.tools.javac.util.Assert;
+import com.univocity.parsers.conversions.DateConversion;
+import com.univocity.parsers.conversions.DoubleConversion;
+
+/**
+ * @author naveen.kasthuri
+ * A convenient container of information represented in a csv row. Instead of handling lists of
+ * arrays of strings, a csv can be fully represented by a list of rows.
+ *
+ * The row object acts as a map from header to a cell. For example: row.get("Make") will return the
+ * value of this row's cell under column "Make". However, each row object does not duplicate the
+ * keys (headers) like a typical Map would: only one copy of the keys is stored in the
+ * {@link RecordMetaData} object. As such, this is a memory efficient, computationally efficient,
+ * convenient container of row objects.
+ *
+ * @see com.univocity.parsers.examples's RowObjectExample for an implementation example
+ */
+public class Row {
+  private final String[] values;
+  private final long lineNumber;
+  private final RecordMetaData recordMetaData;
+  private static final DoubleConversion doubleConversion = new DoubleConversion();
+
+  public Row(String[] row, long lineNumber, RecordMetaData recordMetaData) {
+    this.lineNumber = lineNumber;
+    this.values = row;
+    this.recordMetaData = recordMetaData;
+  }
+
+  public long getLineNumber() {
+    return lineNumber;
+  }
+
+  public Set getHeaders() {
+    return recordMetaData.getHeaders();
+  }
+
+  public String[] getValues() {
+    return values;
+  }
+
+
+  /* Row.get() Methods that are private */
+
+  /**
+   * Returns the value of this row for the given column enum
+   * @return String value
+   */
+  private String get(Enum column) {
+    Assert.checkNonNull(column, "Cannot call get() with a null column");
+    return values[recordMetaData.getIndex(column)];
+  }
+
+  /** Returns a parsed double value or null */
+  private Double getDouble(Enum column) {
+    return doubleConversion.execute(get(column));
+  }
+
+  private Double getDoubleNullIfZero(Enum column) {
+    Double result = doubleConversion.execute(get(column));
+    if (result == 0d) {
+      return null;
+    }
+    return result;
+  }
+
+  /* Row.get() Methods accessible to the public */
+
+  /** Returns the first non-empty non-null string from the list of columns, if one exists */
+  public String get(Enum... columns) {
+    for (Enum column : columns) {
+      String value = get(column);
+      if (value != null && !"".equals(value)) {
+        return value;
+      }
+    }
+    return null;
+  }
```

Comment:
The goal of these and similar variable-argument methods below is similar to the one found in DateConversion.java:
```java
public DateConversion(Date valueIfStringIsNull, String valueIfObjectIsNull, String... dateFormats)
```
It attempts to find the first non-empty column from the list of columns and returns the value.